manual/manual/html_processing/src/process_api.ml

```diff
@@ -0,0 +1,414 @@
+(* Post-processing the HTML of the OCaml API.
+
+   San Vu Ngoc, 2020
+
+   requires Lambdasoup
+
+   cd ..; make web
+
+ *)
+
+open Soup
+open Printf
+open Common
+
+let compiler_libref = ref false
+(* set this to true to process compilerlibref instead of libref *)
+
+let libref = ref ""
+let dst_dir = ref ""
+let src_dir = ref ""
+
+let set_dirs () =
+  libref := if !compiler_libref then "compilerlibref" else "libref";
+  dst_dir :=
+    if !compiler_libref then !libref |> with_dir api_dir
+    else api_dir;
+  src_dir := !libref |> with_dir html_maindir
+
+let () = set_dirs ()
+
+let favicon () = if !compiler_libref then "../favicon.ico" else "favicon.ico"
+
+(* HTML code for the search widget. We don't add the "onchange" event because it
+   forces to click twice to an external link after entering text. *)
+let search_widget with_description =
+  sprintf "<div class=\"api_search\"><input type=\"text\" name=\"apisearch\" id=\"api_search\"
+	 oninput    = \"mySearch(%b);\"
+         onkeypress = \"this.oninput();\"
+         onclick    = \"this.oninput();\"
+	 onpaste    = \"this.oninput();\">
+<img src=\"search_icon.svg\" alt=\"Search\" class=\"svg\" onclick=\"mySearch(%b)\">%s</div>
+<div id=\"search_results\"></div>" with_description with_description
+    (if with_description
+     then {|<span class="search_comment">(search values, type signatures, and descriptions - case sensitive)<div class="search_help"><ul><li>You may search bare values, like <code>map</code>, or indicate the module, like <code>List.map</code>, or type signatures, like <code>int -> float</code>.</li><li>To combine several keywords, just separate them by a space - except if you want to search type signatures: then you must use <strong>two</strong> spaces as separators.</li><li>You may use the special chars <code>^</code> and <code>$</code> to indicate where the matched string should start or end, respectively.</li></ul></div></span>|}
+     else "")
+  |> parse
+
+let process ?(search=true) ~version file out =
+
+  sprintf "Processing %s..." file |> pr;
+  let html = read_file file in
+  let soup = parse html in
+
+  (* Add javascript files *)
+  if search then begin
+    let head = soup $ "head" in
+    create_element "script" ~attributes:["src","search.js"]
+    |> append_child head;
```

Comment:
For me, this is a good example of `|>`: "create an element **and** append it as a child of the head"

Reply:
I agree but sometimes I also like to do eg:
```
mylist |> iter (function a -> .....)
```
especially when the code of the function is long and would hide the list if we put it at the end